### PR TITLE
Support on demand feature views in feature services

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Support on demand feature views in feature services
```

### CODE REVIEW

File: sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py [hunk 1]
```diff
@@ -330,3 +352,62 @@ def test_historical_features(environment, universal_data_sources, full_feature_n
         .reset_index(drop=True)
     )
     assert_frame_equal(actual_df_from_df_entities_for_table, table_from_df_entities)
+
+
+def response_feature_name(feature: str, full_feature_names: bool) -> str:
+    if (
+        feature in {"current_balance", "avg_passenger_count", "lifetime_trip_count"}
```
Comment (adchia) [inline sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py:359]: seems like you only ever call this with conv_rate?

Comment (adchia) [reply sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py:359]: could remove other if blocks if they're not needed

**Review by adchia — APPROVED** [review]

File: sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py (hunk 1 above)
Comment (achals) [reply sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py:359]: Yeah I copied over the one from test_universal_online. Removing.

**Pull request merged**